lib/simplekiq.rb

```diff
@@ -17,5 +17,12 @@ def auto_define_callbacks(batch, args:, job:)
 
       job.define_custom_batch_callbacks(batch) if job.respond_to?(:define_custom_batch_callbacks)
     end
+
+    def format_args(args)
+      return args unless args.is_a?(Array)
+      return args unless args[0].is_a?(Hash)
```

Comment:
technically we should have another early return here if args has more than one element, even if the first element is a hash